frontend/src/api/actions/init.js

```diff
@@ -0,0 +1,150 @@
+    fetchContractTemplates,
+    fetchInstructors,
+    fetchPositions
+} from "./";
+import { setGlobals, globalsSelector } from "../../views/globals/actions";
+
+const toggleMockApi = (getState, newMockAPIValue) => {
+    // in production, ToggleMockApi is a no-op. In development, it actually
+    // does something.
+    /* eslint-disable */
+    if (newMockAPIValue != null && process.env.REACT_APP_DEV_FEATURES) {
+        const mock = require("../mockAPI");
+        const mockAPI = mock.mockAPI;
+        const state = getState();
+
+        if (newMockAPIValue === true) {
+            mockAPI.replaceGlobalFetch();
+        } else {
+            mockAPI.restoreGlobalFetch();
+        }
+    }
+    /* eslint-enable */
+};
+
+const updateGlobals = (getState, newMockAPIValue, newActiveSession) => {
+    const state = getState();
+    let globals = globalsSelector(state);
+
+    if (newMockAPIValue != null) {
+        globals = {
+            ...globals,
+            mockAPI: newMockAPIValue
+        };
+    }
+
+    if (newActiveSession) {
+        globals = {
+            ...globals,
+            activeSession: newActiveSession.id
+        };
+    }
+
+    return setGlobals(globals);
+};
+
+const generateListOfActions = (getState, stage, options) => {
+    // TODO: enable activeUserRole when setActvieUserRole mockAPI/API is implemented
+    const {
+        activeSession = null,
+        // activeUserRole = null,
+        mockAPI = null
+    } = options;
+
+    const onInitialization = [
+        {
+            stage: "toggleMockAPI",
+            function: toggleMockApi,
+            variables: [getState, mockAPI]
+        },
+        {
+            stage: "setActiveUser",
+            action: fetchActiveUser,
+            variables: []
+        },
+        // {
+        //     stage: "setActiveUserRole",
+        //     action: setActiveUserRole,
+        //     variables: [getState, activeUserRole]
+        // },
+        {
+            stage: "fetchSessions",
+            action: fetchSessions,
+            variables: []
+        }
+    ];
+
+    // Only send modify active session when
+    // activeSession is passed in
+    if (activeSession) {
+        onInitialization.push({
+            stage: "setActiveSession",
+            action: setActiveSession,
+            variables: [activeSession]
+        });
+    }
+
+    onInitialization.push({
+        stage: "updateActiveSessionGlobal",
+        action: updateGlobals,
+        variables: [getState, mockAPI, activeSession]
+    });
+
+    while (onInitialization.length > 0) {
+        if (onInitialization[0].stage === stage) {
+            break;
+        }
+
+        onInitialization.shift();
+    }
+    return onInitialization;
+};
+
+const fetchAdditionalData = [
+    fetchApplicants,
+    fetchApplications,
+    fetchAssignments,
+    fetchContractTemplates,
+    fetchInstructors,
+    fetchPositions
+];
+
+export function initFromStage(options, stage = "toggleMockAPI") {
```

Comment:
`stage` before `options`. Stage shouldn't have a default value. Options should (since it's optional). Also, this function needs a comment to explain what it does and what it's for.